tasks/ca.yml

```diff
@@ -55,6 +55,39 @@
   args:
     creates: "{{ ca_ca_dir }}/ca.key"
 
+- name: Check if CA certificate exists
+  ansible.builtin.stat:
+    path: "{{ ca_ca_dir }}/ca.crt"
+  register: ca_certificate
+  when: ca_renew | bool
+
+- name: Verify CA certificate will not expire soon
+  community.crypto.x509_certificate_info:
+    path: "{{ ca_ca_dir }}/ca.crt"
+    valid_at:
+      check_period: "{{ ca_check_valid_time }}"
+  register: ca_expire
+  when:
+    - ca_renew | bool
+    - ca_certificate.stat.exists | bool
+
+- name: Rotate expiring CA certificate
+  ansible.builtin.command: >
+    mv
+    {{ ca_ca_dir }}/ca.crt
+    {{ ca_ca_dir }}/ca.crt.{{ ansible_date_time.iso8601 }}
+  when:
+    - ca_renew | bool
+    - ca_certificate.stat.exists | bool
+    - not ca_expire.valid_at.check_period | bool
+
+- name: Set ca_ca_renewed variable
+  ansible.builtin.set_fact:
+    ca_ca_renewed: "{{ not ca_expire.valid_at.check_period }}"
```

Comment:
I don't really get that part. Isn't `ca_expire.valid_at.check_period` a timeframe? What's the `not` of it?

Reply:
It returns a bool. It will return false if not valid in the given time period, else it will return true.

With `not` I'll invert the bool. I did this to make the variable `ca_ca_renewed` better readable. 

It wouldn't make sense if `ca_ca_renewed` is set to false, although the CA certificate has been renewed.

Reply:
Nice! Now I get it. Thanks.